src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1260,29 +1261,41 @@ internal bool TryInferConversion(IIsPatternOperation operation, out ConversionIn
                 }
 
                 // Handle is pattern operations with constant pattern.
-                if (operation is IIsPatternOperation patternOperation &&
-                    patternOperation.Pattern is IConstantPatternOperation constantPattern)
+                if (operation is IIsPatternOperation isPatternOperation)
                 {
-                    if (constantPattern.Value.ConstantValue.HasValue)
+                    IPatternOperation patternOperation = isPatternOperation.Pattern;
+                    bool direct = true;
+
+                    if (INegatedPatternOperationWrapper.IsInstance(patternOperation))
+                    {
+                        INegatedPatternOperationWrapper negatedPattern = INegatedPatternOperationWrapper.FromOperation(patternOperation);
+                        patternOperation = negatedPattern.Pattern;
+                        direct = false;
+                    }
+
+                    if (patternOperation is IConstantPatternOperation constantPattern)
```

Comment:
else if?

Reply:
No. If pattern is `NegatedPattern`, we set pattern to `NegatedPattern.Pattern`, otherwise we keep the original pattern.
We check if that pattern is `ConstantPattern`, so we either check `is null` or `is not null`.